CMakeLists.txt

```diff
@@ -2,17 +2,16 @@ project(diff-pdf)
 cmake_minimum_required(VERSION 3.0)
 
 set(CMAKE_CXX_STANDARD 17)
-
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
 include(FindPkgConfig)
 
 pkg_search_module(POPPLERGLIB REQUIRED poppler-glib)
-
 add_executable(diff-pdf diff-pdf.cpp)
+
 target_include_directories(diff-pdf PRIVATE
     "${POPPLERGLIB_INCLUDE_DIRS}"
 )
-target_link_libraries(diff-pdf ${POPPLERGLIB_LIBRARIES})
-target_link_directories(diff-pdf PRIVATE
-    ${POPPLERGLIB_LIBRARY_DIRS}
-)
-install(TARGETS diff-pdf)
+
+target_link_libraries(diff-pdf ${POPPLERGLIB_LIBRARIES} stdc++fs)
```

Comment:
Adding `stdc++fs` here breaks the build on macos. Is this really needed ?

If this is the case, then I would change this to:

```
target_link_libraries(diff-pdf ${POPPLERGLIB_LIBRARIES}
if(CMAKE_SYSTEM_NAME MATCHES "Linux")
    target_link_libraries(diff-pdf stdc++fs)
endif()
```

Reply:
it's been so long since I looked at this that I can't totally remember, but i'm about 85% sure it was necessary... with the strong caveat that this is the first time i have ever looked at let alone edited a CMake build file.

i will be dealing w/the project this change is for in the next few weeks so I will probably have time to test out your suggested change.